pandas/core/indexes/base.py

```diff
@@ -5206,8 +5206,29 @@ def sort_values(
         >>> idx.sort_values(ascending=False, return_indexer=True)
         (Int64Index([1000, 100, 10, 1], dtype='int64'), array([3, 1, 0, 2]))
         """
+        from pandas.core.indexes.range import RangeIndex
+
         idx = ensure_key_mapped(self, key)
 
+        if isinstance(self, RangeIndex):
```

Comment:
instead of this you can override `.sort_values` in RangeIndex to handle this

Reply:
There is a @final in the original function. Should I remove it?

Reply:
yes

Reply:
Done